tests/gatewayapitests/graphlayer/gateway_test.go

```diff
@@ -340,15 +340,13 @@ func TestGatewayTLSToNoTLS(t *testing.T) {
 	}, 30*time.Second).Should(gomega.Equal(true))
 
 	g.Eventually(func() bool {
-		found, _ := objects.SharedAviGraphLister().Get(modelName)
-		return found
-	}, 25*time.Second).Should(gomega.Equal(true))
+		_, aviModel = objects.SharedAviGraphLister().Get(modelName)
```

Comment:
@arihantg: It will be good to check presence of model/ nodes in a cache with `for wait` condition instead of specifying individual condition in that node. We can increase timeout if node is not getting populated.

Reply:
model is already found in the check at line 318 . we have to check precisely this condition since other conditions will be same before the gatewayupdate at line 331